der/derive/src/enumerated.rs

```diff
@@ -19,13 +19,16 @@ pub(crate) struct DeriveEnumerated {
     /// Value of the `repr` attribute.
     repr: Ident,
 
+    /// Whether or not to tag the enum as an integer
+    integer: bool,
+
```

Comment:
This seems like an inextensible way to override the tag of an `Enumerated`, so why not just add a `#[asn1(tag = "...")]` argument instead that lets you override the `ENUMERATED` tag however you want? Then it could potentially work for more than just `INTEGER`.